cg/meta/demultiplex/demux_post_processing.py

```diff
@@ -162,10 +162,20 @@ def add_sequencing_metrics_to_statusdb(
         self, sample_lane_sequencing_metrics: List[SampleLaneSequencingMetrics]
     ) -> None:
         for metric in sample_lane_sequencing_metrics:
-            if not self.metric_exists_in_status_db(metric):
+            if not self.metric_exists_in_status_db(metric) and self.metric_has_sample_in_statusdb(
```

Comment:
Nice 👍  We could split this long if statement into several lines to potentially make it more readable, you determine if it is worth:

```python
    for metric in sequencing_metrics:
        metric_exists = self.is_metric_in_database(metric)
        sample_exists = self.is_sample_in_database(metric)
        
        if not metric_exists and sample_exists:
            self.add_metric_to_database(metric)
    
    self.status_db.session.commit()
```

Reply:
Yes it does not do the line skipping, i agree fixed it